### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Kevin-Solomon**
# Bugs fixed 
- [x] Disabled addToCart button after one click to avoid multiple requests sent to backend.
- [x] Passed missing rating prop to card components.
- [x] Change wishlist icon.
- [x] Add a component to display if there is no item present in the cart or wishlist.
- [x] Made Cart and wishlist back to empty after logout.
- [x] Add useEffect to initialize cart items if user login.

### CODE REVIEW

**Comment by AnkurChunekar** [comment]
kevin pr
- As you can see if I click on wishlist icon more than 1 time it adds same products again. please look into this.
- Also the wishlist icon quantity is not correct for the above bug you can fix that too.

kevin pr
- I have encountered the same error for cart when I increase item qty and decrease it quickly.

File: src/components/Card/Card.css [hunk 1]
```diff
@@ -84,8 +88,8 @@
 }
 .card-content .btn {
   margin: 0;
-  background-color: #2563eb;
-  color: white;
+  /* background-color: #2563eb;
+  color: white; */
```
Comment (AnkurChunekar) [inline src/components/Card/Card.css]: remove unnecessary comments

File: src/components/Card/Card.css [hunk 2]
```diff
@@ -111,3 +115,23 @@
...
+}
+
+.card-footer-btn > * {
+  flex-basis: 50%;
+}
+.outline-primary {
+  border: 1px solid var(--primary-color);
+}
+.icon-btn {
+  background-color: aqua;
```
Comment (AnkurChunekar) [inline src/components/Card/Card.css]: Try to have consistent line spacing

File: src/pages/SingleProductPage/components/ProductContainer/ProductContainer.jsx [hunk 3]
```diff
@@ -3,20 +3,25 @@ import { useParams } from 'react-router-dom';
...
+          response.data.products.find(item => item._id === params.productId)
+        );
+      } catch (err) {
+        console.log(err);
+      }
     }
     getProducts();
   }, []);
-
+  console.log(data);
```
Comment (AnkurChunekar) [inline src/pages/SingleProductPage/components/ProductContainer/ProductContainer.jsx]: remove console logs

File: src/reducer/cartReducer.js [hunk 4]
```diff
@@ -7,6 +7,8 @@ const DELETE_ITEM = 'DELETE_ITEM';
 const DELETE_EXTRA_ITEM = 'DELETE_EXTRA_ITEM';
 const DECREASE_QUANTITY = 'DECREASE_QUANTITY';
 const INCREASE_QUANTITY = 'INCREASE_QUANTITY';
+const RESET = 'RESET';
+const INITIALIZE = 'INITIALIZE';
```
Comment (AnkurChunekar) [inline src/reducer/cartReducer.js:11]: This looks very neat and clean, great work here

**Review by AnkurChunekar — CHANGES_REQUESTED** [review]
Everything else looks good, Great work @Kevin-Solomon

File: src/pages/WishList/WishList.jsx [hunk 5]
```diff
@@ -2,28 +2,31 @@ import React from 'react';
...
 import Card from '../../components/Card/Card';
 import './WishList.css';
-import axios from 'axios';
-import { useState, useEffect } from 'react';
-import { getByTitle } from '@testing-library/react';
+import { NoItemFound } from '../../components/NoItemFound/NoItemFound';
 import { useWishlist } from '../../context/wishlist/wishlistContext';
 function WishList() {
   const { wishlistState } = useWishlist();
+  console.log(wishlistState);
```
Comment (DevRaahul) [inline src/pages/WishList/WishList.jsx]: remove this

**Comment by AnkurChunekar** [comment]
kevin pr
- Hey kevin i have encountered the same error for cart when I increase item qty and decrease it quickly.

**Review by DevRaahul — APPROVED** [review]

**Review by DevRaahul — COMMENTED** [review]
https://drive.google.com/file/d/1oGhhMDp2SH_GwT-KHmW1ELtSuSziQ2Bz/view

One suggestion here is when a user clicked on the view item that time if the scroll goes to its initial position then we can directly see the product. It will enhance the user experience i guess.
Refer this - https://drive.google.com/file/d/1oGhhMDp2SH_GwT-KHmW1ELtSuSziQ2Bz/view

**Review by AnkurChunekar — APPROVED** [review]

**Pull request merged**